### PR TITLE
Use python native warnings

### PR DESCRIPTION
**Pull request by bartv**

### CODE REVIEW

File: plugins/__init__.py [hunk 1]
```diff
@@ -44,7 +43,7 @@
 resource_states: dict[str, dict] = dict()
 
 
-class InmantaDeprecationWarning(DeprecationWarning, InmantaWarning):
+class InmantaDeprecationWarning(DeprecationWarning):
```
Comment (edvgui) [inline plugins/__init__.py:46]: I guess we can simply get rid of the class?

**Review by edvgui — CHANGES_REQUESTED** [review]

File: plugins/__init__.py (hunk 1 above)
Comment (bartv) [reply plugins/__init__.py:46]: true. Will remove it then

File: tests/features/test_config_tree.py [hunk 2]
```diff
@@ -141,12 +141,12 @@ def test_deprecated_config(project: Project) -> None:
 
     desired_logs = {
         (
-            "inmanta.warnings         WARNING InmantaDeprecationWarning: The usage of config '' at "
+            "inmanta.warnings         WARNING DeprecationWarning: The usage of config '' at "
```
Comment (edvgui) [inline tests/features/test_config_tree.py:144]:
```suggestion
            "py.warnings              WARNING DeprecationWarning: The usage of config '' at "
```

File: tests/features/test_config_tree.py [hunk 3]
```diff
@@ -141,12 +141,12 @@ def test_deprecated_config(project: Project) -> None:
...
             f"{project._test_project_dir}/main.cf:3 is deprecated"
         ),
         (
             # prior to https://github.com/inmanta/inmanta-core/commit/64798acc8abcc3b7b31ab657f1d04e1974209d6f
-            "inmanta.warnings         WARNING InmantaDeprecationWarning: The usage of config '' at "
+            "inmanta.warnings         WARNING DeprecationWarning: The usage of config '' at "
```
Comment (edvgui) [inline tests/features/test_config_tree.py:149]:
```suggestion
            "py.warnings              WARNING DeprecationWarning: The usage of config '' at "
```

**Review by edvgui — APPROVED** [review]

File: tests/features/test_config_tree.py [hunk 4]
```diff
@@ -141,12 +141,12 @@ def test_deprecated_config(project: Project) -> None:
...
+            "py.warnings              WARNING DeprecationWarning: The usage of config '' at "
             f"{project._test_project_dir}/main.cf:3 is deprecated"
         ),
         (
             # prior to https://github.com/inmanta/inmanta-core/commit/64798acc8abcc3b7b31ab657f1d04e1974209d6f
-            "inmanta.warnings         WARNING InmantaDeprecationWarning: The usage of config '' at "
+            "py.warnings              WARNING DeprecationWarning: The usage of config '' at "
             "./main.cf:3 is deprecated"
         ),
     }
```
Comment (edvgui) [inline tests/features/test_config_tree.py:152]:
```python
    desired_logs = {
        (
            f"py.warnings              WARNING {project._test_project_dir}/libs/terraform/plugins/__init__.py:469: "
            f"DeprecationWarning: The usage of config '' at {project._test_project_dir}/main.cf:3 is deprecated"
        ),
        (
            f"py.warnings              WARNING {project._test_project_dir}/libs/terraform/plugins/__init__.py:469: "
            "DeprecationWarning: The usage of config '' at ./main.cf:3 is deprecated"
        ),
    }
```

This should work (I tested it this time)

**Comment by edvgui** [comment]
The PR of @FloLey actually also adds a changelog entry and makes a version bump, I am still not clear what the procedure is for updating modules in core, but I think we at least need the changelog entry here

**Review by edvgui — CHANGES_REQUESTED** [review]
The PR of @FloLey actually also adds a changelog entry and makes a version bump, I am still not clear what the procedure is for updating modules in core, but I think we at least need the changelog entry here

**Comment by FloLey** [comment]
I always follow this doc for adding changes to modules: https://internal.inmanta.com/developer/tasks/commiting_changes_modules.html

**Pull request merged**